framework/test-roborazzi/src/androidMain/kotlin/com/cramsan/framework/test/roborazzi/MultiplatformPreviewTester.kt

```diff
@@ -0,0 +1,103 @@
+import com.github.takahirom.roborazzi.ExperimentalRoborazziApi
+import com.github.takahirom.roborazzi.RoborazziActivity
+import com.github.takahirom.roborazzi.captureRoboImage
+import com.github.takahirom.roborazzi.registerRoborazziActivityToRobolectricIfNeeded
+import org.junit.rules.RuleChain
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+import org.robolectric.Robolectric
+import org.robolectric.util.Logger
+import sergio.sastre.composable.preview.scanner.jvm.JvmAnnotationInfo
+import sergio.sastre.composable.preview.scanner.jvm.JvmAnnotationScanner
+
+@OptIn(ExperimentalRoborazziApi::class)
+class MultiplatformPreviewTester : ComposePreviewTester<JUnit4TestParameter<JvmAnnotationInfo>> {
+    override fun options(): ComposePreviewTester.Options = super.options().copy(
+        testLifecycleOptions = ComposePreviewTester.Options.JUnit4TestLifecycleOptions(
+            composeRuleFactory = {
+                @Suppress("UNCHECKED_CAST")
+                createAndroidComposeRule<RoborazziActivity>()
+                    as AndroidComposeTestRule<ActivityScenarioRule<out ComponentActivity>, *>
+            },
+            testRuleFactory = { composeTestRule ->
+                RuleChain.outerRule(
+                    object : TestWatcher() {
+                        override fun starting(description: Description?) {
+                            super.starting(description)
+                            registerRoborazziActivityToRobolectricIfNeeded()
+                        }
+                    }
+                )
+                    .around(composeTestRule)
+            }
+        )
+    )
+
+    override fun testParameters(): List<JUnit4TestParameter<JvmAnnotationInfo>> {
+        val options = options()
+        val annotations = listOf(
+            "org.jetbrains.compose.ui.tooling.preview.Preview",
+            "androidx.compose.ui.tooling.preview.Preview"
+        )
+        return annotations.map { JvmAnnotationScanner(it) }
+            .map {
+                it
+                    .scanPackageTrees(*options.scanOptions.packages.toTypedArray())
+                    .includePrivatePreviews()
+                    .getPreviews()
+            }
+            .flatten()
+            .map {
+                JUnit4TestParameter(
+                    (
+                        options.testLifecycleOptions as
+                            ComposePreviewTester.Options.JUnit4TestLifecycleOptions
+                        ).composeRuleFactory,
+                    it
+                )
+            }
+    }
+
+    override fun test(testParameter: JUnit4TestParameter<JvmAnnotationInfo>) {
+        // Setting the context to be able to access resources
+        setupAndroidContextProvider()
+
+        val preview = testParameter.preview
+        testParameter.composeTestRule.setContent {
+            preview()
+        }
+        testParameter.composeTestRule.onRoot()
```

Comment:
I believe if you have to pass a meaningful name to the captureRoboImage method, for instance:   `captureRoboImage(preview.toString())` but you could try by combining other preview params like
`captureRoboImage(preview.className + preview.methodName)`